app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -214,50 +212,57 @@ class SearchActivity : AppCompatActivity() {
         return Gson().fromJson(json, Array<Track>::class.java)
     }
 
-    private fun searchTracks() {
-        iTunesService.getTracks(binding.inputEditText.text.toString())
-            .enqueue(object : Callback<TracksResponse> {
-                override fun onResponse(
-                    call: Call<TracksResponse>,
-                    response: Response<TracksResponse>
-                ) {
-                    when (response.code()) {
-                        200 -> {
-                            if (response.body()?.results?.isNotEmpty() == true) {
-                                tracks.clear()
-                                binding.placeholderNoSearchResults.isVisible = false
-                                binding.placeholderServerError.isVisible = false
-                                tracks.addAll(response.body()?.results!!)
-                                searchAdapter.notifyDataSetChanged()
-                            } else {
+    private fun searchRequest() {
+        if (searchText.isNotEmpty()) {
+            binding.placeholderNoSearchResults.isVisible = false
+            binding.placeholderServerError.isVisible = false
+            binding.tracksRecyclerView.isVisible = false
+            binding.searchProgressBar.isVisible = true
+            iTunesService.getTracks(binding.inputEditText.text.toString())
+                .enqueue(object : Callback<TracksResponse> {
+                    override fun onResponse(
+                        call: Call<TracksResponse>,
+                        response: Response<TracksResponse>
+                    ) {
+                        binding.searchProgressBar.isVisible = false
+                        when (response.code()) {
+                            200 -> {
+                                if (response.body()?.results?.isNotEmpty() == true) {
+                                    tracks.clear()
+                                    binding.placeholderNoSearchResults.isVisible = false
+                                    binding.placeholderServerError.isVisible = false
+                                    tracks.addAll(response.body()?.results!!)
+                                    searchAdapter.notifyDataSetChanged()
```

Comment:
```suggestion
                                    searchAdapter.notifyDataSetChanged()
                                    binding.tracksRecyclerView.isVisible = true
```
Успешный результат запроса не отображается, потому что список скрыт
Лучше полностью отрефакторить этот спагетти-код